### PR TITLE
Add StreamSizeTrigger

### PR DESCRIPTION
**Pull request by ecfm**

### CODE REVIEW

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/trigger/StreamSizeTrigger.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+ */
+
+package co.cask.cdap.internal.app.runtime.schedule.trigger;
+
+import co.cask.cdap.proto.id.StreamId;
+
+/**
+ * A Trigger that schedules a ProgramSchedule, based upon data availability in a stream.
+ */
+public class StreamSizeTrigger {
```
Comment (chtyim) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/trigger/StreamSizeTrigger.java:24]: Why this one doesn't extends from `Trigger` class?

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/trigger/StreamSizeTrigger.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    }
+    StreamSizeTrigger that = (StreamSizeTrigger) o;
+    return dataTriggerMB == that.dataTriggerMB && streamName.equals(that.streamName);
+  }
+
+  @Override
+  public int hashCode() {
+    int result = streamName.hashCode();
+    result = 31 * result + dataTriggerMB;
+    return result;
```
Comment (chtyim) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/trigger/StreamSizeTrigger.java:57]: Use `Objects.hash`.

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/trigger/StreamSizeTrigger.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+public class StreamSizeTrigger {
+  private final StreamId streamName;
+  private final int dataTriggerMB;
+
+  public StreamSizeTrigger(StreamId streamName, int dataTriggerMB) {
+    this.streamName = streamName;
+    this.dataTriggerMB = dataTriggerMB;
+  }
+
+  public StreamId getStreamName() {
```
Comment (chtyim) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/trigger/StreamSizeTrigger.java:33]: better call the method `getStreamId`.

**Review by chtyim — COMMENTED** [review]
What is this PR for? It just add a new class without any usage of it?

**Comment by ecfm** [comment]
Temporarily close this PR until refactoring CoreSchedulerService

**Pull request closed without merging**